### PR TITLE
CDC Contract testing

### PR DESCRIPTION
**Pull request by SachinNaikHMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/contractTest/resources/application-contract.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+targetHost: ${TEST_URL:hmi-apim-svc-test.azure-api.net}
+targetSubscriptionKey: ${TEST_SUBSCRIPTION_KEY:587c56f59a604054b1ce73dac0ea5b28}
```
Comment (karanotts) [inline src/contractTest/resources/application-contract.yaml:2]: Please do not push subscription keys or any other secrets to github

File: .env [hunk 2]
```diff
@@ -1 +1,5 @@
 SERVER_PORT=4550
```
Comment (pats-john) [inline .env:1]: Most likely this .env file would not be required and values would only be required to be pushed in from the Command Line.Let this be as such for now and we can see what Dev Ops require going forward.

File: build.gradle [hunk 3]
```diff
@@ -336,8 +369,40 @@ dependencies {
 
   smokeTestImplementation sourceSets.main.runtimeClasspath
 
-  integrationTestImplementation sourceSets.main.runtimeClasspath
```
Comment (pats-john) [inline build.gradle:339]: Why are these 2 lines of Code removed.Integration test should stay put as they are...This task should only put in values,variables and code for contractTests.

File: build.gradle [hunk 4]
```diff
@@ -372,3 +437,44 @@ bootJar {
...
+  publish {
+    pactDirectory = 'target/pacts'
+    pactBrokerUrl = System.getenv("PACT_BROKER_FULL_URL") ?: 'http://localhost:80'
+    tags = [System.getenv("PACT_BRANCH_NAME") ?: 'Dev']
+    version = project.pactVersion
+  }
+
+  serviceProviders {
+
+    SandL_HEARING_API {
```
Comment (pats-john) [inline build.gradle:467]: Rename it to SandL....Because SandL does have a flat concept of Hearing....

**Comment by pats-john** [comment]
Out of Date Brannch and this feature was later merged to master through another branch so closed and deleted brnach.

**Pull request closed without merging**